### PR TITLE
Renaming and burn method

### PR DESCRIPTION
**Pull request by MoMannn**

### CODE REVIEW

File: contracts/tokens/NFTokenEnumerable.sol [hunk 1]
```diff
@@ -52,6 +52,29 @@ contract NFTokenEnumerable is NFToken, ERC721Enumerable {
...
+   * @param _tokenId ID of the NFToken to be burned.
+   */
+  function _burn(address _owner,
+                 uint256 _tokenId)
+    internal
+  {
+    super._burn(_owner, _tokenId);
+
+    uint256 tokenIndex = idToIndex[_tokenId];
+    uint256 lastTokenIndex = tokens.length.sub(1);
```
Comment (lknix) [inline contracts/tokens/NFTokenEnumerable.sol]: Can we add `assert(tokens.length > 0);` here?

File: contracts/tokens/NFTokenEnumerable.sol [hunk 2]
```diff
@@ -52,6 +52,29 @@ contract NFTokenEnumerable is NFToken, ERC721Enumerable {
...
+    uint256 lastToken = tokens[lastTokenIndex];
+
+    tokens[tokenIndex] = lastToken;
+    tokens[lastTokenIndex] = 0;
+
+    tokens.length--;
+    idToIndex[_tokenId] = 0;
+    idToIndex[lastToken] = tokenIndex;
```
[the remark this answers is not in the document]
Comment (MoMannn) [reply contracts/tokens/NFTokenEnumerable.sol]: It is not the same.. removeNFToken does the removal from token owner list, burn does from the token list..

File: contracts/tokens/NFTokenEnumerable.sol (hunk 1 above)
Comment (MoMannn) [reply contracts/tokens/NFTokenEnumerable.sol]: Will do.

File: contracts/tokens/NFTokenEnumerable.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (lknix) [reply contracts/tokens/NFTokenEnumerable.sol]: Yeah, you're right! Completely mixed the two :) When we burn the token, why don't we update the owner token list, too? I guess it has something to do with keeping the token footprint.

**Comment by lknix** [comment]
One comment left (which is just for my curiosity). LGTM, go ahead and merge

File: contracts/tokens/NFTokenEnumerable.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (MoMannn) [reply contracts/tokens/NFTokenEnumerable.sol]: We actually do update it. We call super._burn which calls removeNFToken :)

**Pull request merged**